src/mam/mod_mam_riak_timed_arch_yz.erl

```diff
@@ -203,22 +209,34 @@ remove_bucket(Bucket) ->
     {ok, Keys} = mongoose_riak:list_keys(Bucket),
     [mongoose_riak:delete(Bucket, Key) || Key <- Keys].
 
-archive_message(Host, MessID, LocJID, RemJID, SrcJID, Packet, Type) ->
+
+%% PM:
+%%  * LocJID - From/To JID depending on direction of message (single message is saved in two directions)
+%%  * RemJID - To/From JID depending on direction of message (single message is saved in two directions)
+%%  * SrcJID - "Real" sender JID
+%%  * OwnerJID - Same as LocJID
+%% MUC / MUC Light:
+%%  * LocJID - MUC/MUC Light room's JID
+%%  * RemJID - Nickname of JID of destination
+%%  * SrcJID - Full JID of user within room (room@domain/user)
+%%  * OwnerJID - "Real" sender JID (not room specific)
+archive_message(Host, MessID, LocJID, RemJID, SrcJID, OwnerJID, Packet, Type) ->
```

Comment:
I have a feeling that if we need to explain what variable means, depending on some context, we could decompose the logic. However, I understand thats implication of working with legacy code

Reply:
This would definitely be solved by a refactor (in the future), where we could replace these arguments with parameter map. Indeed now it's confusing as ***.

Reply:
Yes, indeed this is very confusing. But changing this is this module only will only make things worse, as other MAM backends use exactly the same wording. 
@ludwikbukowski I actually tried to provide descriptions without context, because those arguments seem to be abstract enough (in the end, whole logic of this function doesn't do anything specific depending on context), but I simply failed to figure out the proper wording. Still, I believe that having those definitions here even context-specific is better then not having it and guessing each time we develop an extension.